api/sbom.proto

```diff
@@ -150,6 +151,22 @@ message Tool {
     string vendor = 3;
 }
 
+message DocumentType {
+    optional SBOMType type = 1;
+    optional string name = 2;
+    optional string description = 3;
+
+    enum SBOMType {
+        OTHER = 0;
+        DESIGN = 1;
+        SOURCE = 2;
+        BUILD = 3;
+        ANALYZED = 4;
+        DEPLOAYED = 5;
+        RUNTIME = 6;
+    }
```

Comment:
To keep the promise of lossless ingestion, my suggestion here is to also add the lifecycle phases that don't map to the community types.

Ingesting a cdx doc and rewriting it again should be lossless, so we need those types. Data loss (and the `OTHER` type) would only come into play when using the spdx sinks or older cdx.

Based on your mapping function and on the previous discussions on lifecycles vs sbom types, I think this should do (also watch out for the typo on `DEPLOYED`). Also note the mapping notes on the function below.

```suggestion
    enum SBOMType {
        OTHER = 0;
        DESIGN = 1; // design
        SOURCE = 2; // pre-build
        BUILD = 3; // build
        ANALYZED = 4; // post-build
        DEPLOYED = 5; // operations
        RUNTIME = 6;
        DISCOVERY = 7;  // CDX, no clear match
        DECOMISSION = 8;   // CDX, no clear match
    }
```

Reply:
Thank you for your help! I was having trouble converting to/from CDX according to the CISA document. After narrowing it down to a minimum of 6, I agree with your suggestion. we should include the CDX specific enums (discovery/decomission)

Reply:
@puerco so what do we need "OTHER" for? if we go "inclusive" (adding "DISCOVERY" and "DECOMISSION"). 
In my implementation, if a "phase" is OTHER, I still use the "phase name" (eg. "discovery") as a phase. 

Looking at https://github.com/spdx/spdx-3-model/blob/f73b70d8cd86ead09902937cbfdfdcd93af3ef1c/model/Core/Vocabularies/LifecycleScopeType.md#L4

SPDX 3 lifecycles are these:
```
design: TODOdescription
build: TODOdescription
development: TODOdescription
test: TODOdescription
runtime: TODOdescription
other: TODOdescription
```

I assume we should "save" the OTHER prop for "non-phase" in CDX, so we can go from/to CDX1.5/SPDX3.
However, there is a big difference between usage for "lifecycle" CDX in metadata, and SPDX as a relationship type. We will be able to translate those, but I'm pretty sure we would lose the "description" when going from CDX to SPDX.

CDX 1.5
```json
 "lifecycles": [
      {
        "phase": "build"
      },
      {
        "name": "platform-integration-testing",
        "description": "Integration testing specific to the runtime platform"
      }
    ],
```

I suggest we discuss this briefly in today's Cohort meeting

Reply:
Yeah in my view, `OTHER` should be the default when unknown data makes it into the field (which should never happen via the unserializers). It should be the value populated when using the struct option in CDX. In other enums we use `UNKNOWN` but here I think OTHER is fine as it can also point to.

Regarding the SPDX3 values, you are confusing the lifecycles of the build profile with the SBOM Type. This is the one you should be looking at:

https://github.com/spdx/spdx-3-model/blob/f73b70d8cd86ead09902937cbfdfdcd93af3ef1c/model/Software/Vocabularies/SbomType.md

All the build profile stuff is out of the scope of protobom, so don't worry about it :)

Reply:
No no, don't do that. If the lifecycle is in the enum and you move it to the opt2 then you are effectively modifying the document before you are even finished reading it.

When ingesting data into enums, we grow the options to handle the vocabularies of the formats to not lose any data. See the hash algorithms, for example: There are common algorithms and then there are others that are specific to SPDX. Regardless they all exist in the enum.

Reply:
Oh ok, Ill do that then!